pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -3883,85 +3885,2699 @@ func Test_server_MultiInsert(t *testing.T) {
+	)
+	var (
+		ip = net.LoadLocalIP() // agent ip address
+
+		// default NGT configuration for test
+		defaultIntSvcCfg = &config.NGT{
+			Dimension:    intVecDim,
+			DistanceType: ngt.Angle.String(),
+			ObjectType:   ngt.Uint8.String(),
+			KVSDB:        &config.KVSDB{},
+			VQueue:       &config.VQueue{},
+		}
+		defaultF32SvcCfg = &config.NGT{
+			Dimension:    f32VecDim,
+			DistanceType: ngt.Angle.String(),
+			ObjectType:   ngt.Float.String(),
+			KVSDB:        &config.KVSDB{},
+			VQueue:       &config.VQueue{},
+		}
+		defaultSvcOpts = []service.Option{
+			service.WithEnableInMemoryMode(true),
+		}
+	)
+
+	genF32Vec := func(dist vector.Distribution, num int, dim int) [][]float32 {
+		generator, _ := vector.Float32VectorGenerator(dist)
+		return generator(num, dim)
+	}
+	genIntVec := func(dist vector.Distribution, num int, dim int) [][]float32 {
+		generator, _ := vector.Uint8VectorGenerator(dist)
+		ivecs := generator(num, dim)
+		result := make([][]float32, num)
+
+		for j, ivec := range ivecs {
+			vec := make([]float32, dim)
+			for i := 0; i < dim; i++ {
+				vec[i] = float32(ivec[i])
+			}
+			result[j] = vec
+		}
+		return result
+	}
+
+	genMultiInsertReq := func(t objectType, dist vector.Distribution, num int, dim int) *payload.Insert_MultiRequest {
+		var vecs [][]float32
+		switch t {
+		case Float:
+			vecs = genF32Vec(dist, num, dim)
+		case Uint8:
+			vecs = genIntVec(dist, num, dim)
+		}
+
+		req := &payload.Insert_MultiRequest{
+			Requests: make([]*payload.Insert_Request, num),
+		}
+		for i, vec := range vecs {
+			req.Requests[i] = &payload.Insert_Request{
+				Vector: &payload.Object_Vector{
+					Id:     "uuid-" + strconv.Itoa(i+1),
+					Vector: vec,
+				},
+			}
+		}
+
+		return req
+	}
+
+	// generate MultiInsert request with the same vector
+	genSameVecMultiInsertReq := func(num int, vec []float32) *payload.Insert_MultiRequest {
+		req := &payload.Insert_MultiRequest{
+			Requests: make([]*payload.Insert_Request, num),
+		}
+		for i := 0; i < num; i++ {
+			req.Requests[i] = &payload.Insert_Request{
+				Vector: &payload.Object_Vector{
+					Id:     "uuid-" + strconv.Itoa(i+1),
+					Vector: vec,
+				},
+			}
+		}
+
+		return req
+	}
+
+	genObjectLocations := func(num int, name string, ip string) *payload.Object_Locations {
+		result := &payload.Object_Locations{
+			Locations: make([]*payload.Object_Location, num),
+		}
+
+		for i := 0; i < num; i++ {
+			result.Locations[i] = &payload.Object_Location{
+				Name: name,
+				Uuid: "uuid-" + strconv.Itoa(i+1),
+				Ips:  []string{ip},
+			}
+		}
+		return result
+	}
+
 	defaultCheckFunc := func(w want, gotRes *payload.Object_Locations, err error) error {
-		if !errors.Is(err, w.err) {
-			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		if !w.haveErr {
+			if !errors.Is(err, w.err) {
+				return errors.Errorf("got_error: \"%v\",\n\t\t\t\twant: \"%v\"", err, w.err)
+			}
```

Comment:
IMO, I'd like to avoid using this branch.
Why you do not want to check the error.

Reply:
the reason is written here.
https://github.com/vdaas/vald/pull/1612/files#r876765568